File: tatala-java/src/com/qileyuan/tatala/socket/to/helper/IntArrayMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class IntArrayMapHelper {

	public static int getByteArrayLength(Map<String, int[]> intArrayMap){
		int blength = 0;
		Iterator<String> keyIt = intArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			int[] values = intArrayMap.get(key);
			blength += TransferUtil.getLengthOfIntArray(values);
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, int[]> intArrayMap, TransferOutputStream touts) {
		Iterator<String> keyIt = intArrayMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_INTARRAY);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			int[] values = intArrayMap.get(key);
			touts.writeIntArray(values);
		}
	}
}

File: tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/IntArrayMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class IntArrayMapHelper {
        public static int getByteArrayLength(Dictionary<String, int[]> intArrayMap) {
            int blength = 0;
            foreach (KeyValuePair<String, int[]> item in intArrayMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                int[] values = item.Value;
                blength += TransferUtil.getLengthOfIntArray(values);
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, int[]> intArrayMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, int[]> item in intArrayMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_INTARRAY);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                int[] values = item.Value;
                touts.writeIntArray(values);
            }
        }
    }
}